pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go

```diff
@@ -0,0 +1,181 @@
+		image:    image,
+		command:  command,
+		port:     port,
+	}
+}
+
+// InjectFS implements cmdutil.Scaffolder
+func (s *apiScaffolder) InjectFS(fs machinery.Filesystem) {
+	s.fs = fs
+}
+
+// Scaffold implements cmdutil.Scaffolder
+func (s *apiScaffolder) Scaffold() error {
+	fmt.Println("Writing scaffold for you to edit...")
+
+	// Load the boilerplate
+	boilerplate, err := afero.ReadFile(s.fs.FS, hack.DefaultBoilerplatePath)
+	if err != nil {
+		return fmt.Errorf("error scaffolding API/controller: unable to load boilerplate: %w", err)
+	}
+
+	if err := s.config.UpdateResource(s.resource); err != nil {
+		return fmt.Errorf("error updating resource: %w", err)
+	}
+
+	// Initialize the machinery.Scaffold that will write the files to disk
+	scaffold := machinery.NewScaffold(s.fs,
+		machinery.WithConfig(s.config),
+		machinery.WithBoilerplate(string(boilerplate)),
+		machinery.WithResource(&s.resource),
+	)
+
+	if err := scaffold.Execute(
+		&api.Types{Command: s.command, Port: s.port},
+		&api.Group{},
+	); err != nil {
+		return fmt.Errorf("error scaffolding APIs: %v", err)
+	}
+
+	controller := &controllers.Controller{ControllerRuntimeVersion: scaffolds.ControllerRuntimeVersion}
+	if err := scaffold.Execute(
+		&controllers.SuiteTest{},
+		controller,
+	); err != nil {
+		return fmt.Errorf("error scaffolding controller: %v", err)
+	}
+
+	controllerPath := controller.Path
+	if err := util.ReplaceInFile(controllerPath, "//TODO: scaffold container",
+		fmt.Sprintf(containerTemplate,
+			s.image,                          // value for the image
+			strings.ToLower(s.resource.Kind), // value for the name of the container
+		),
+	); err != nil {
+		return fmt.Errorf("error scaffolding container in the controller: %v", err)
+	}
+
+	// Scaffold the command if informed
+	if len(s.command) > 0 {
+		err := util.InsertCode(controllerPath, `SecurityContext: &corev1.SecurityContext{
+							RunAsNonRoot:             &[]bool{true}[0],
+							AllowPrivilegeEscalation: &[]bool{false}[0],
+							Capabilities: &corev1.Capabilities{
+								Drop: []corev1.Capability{
+									"ALL",
+								},
+							},
+						},`, commandTemplate)
+		if err != nil {
+			return fmt.Errorf("error scaffolding the command in the controller: %v", err)
+		}
+	}
+
+	// Scaffold the port if informed
+	if len(s.port) > 0 {
+		err := util.InsertCode(controllerPath, `SecurityContext: &corev1.SecurityContext{
+							RunAsNonRoot:             &[]bool{true}[0],
+							AllowPrivilegeEscalation: &[]bool{false}[0],
+							Capabilities: &corev1.Capabilities{
+								Drop: []corev1.Capability{
+									"ALL",
+								},
+							},
+						},`, fmt.Sprintf(portTemplate, strings.ToLower(s.resource.Kind)))
+		if err != nil {
+			return fmt.Errorf("error scaffolding container port in the controller: %v", err)
+		}
+	}
+
+	if err := scaffold.Execute(
+		&samples.CRDSample{Command: s.command, Port: s.port},
+	); err != nil {
+		return fmt.Errorf("error updating config/samples: %v", err)
+	}
+
```

Comment:
We need to check the main.go:

```go vet ./...
# sigs.k8s.io/kubebuilder/testdata/project-v3-with-deploy-image
vet: ./main.go:88:13: undeclared name: examplecomv1alpha1
```
What is missing to update?
How the create API without the plugin works here?

Reply:
See that the e2e fails at the same point : https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kubebuilder/2717/pull-kubebuilder-e2e-k8s-1-19-16/1538508442782666752#1:build-log.txt%3A873